rsconf/component/mongod.py

```diff
@@ -0,0 +1,55 @@
+# -*- coding: utf-8 -*-
+"""create nginx configuration
+
+:copyright: Copyright (c) 2017 RadiaSoft LLC.  All Rights Reserved.
+:license: http://www.apache.org/licenses/LICENSE-2.0.html
+"""
+from pykern import pkio
+from pykern.pkcollections import PKDict
+from pykern.pkdebug import pkdp
+from rsconf import component
+
+_DATA_DIR = "db"
+
+_LOG_DIR = "log"
+
+_MONGO_CONF_F = "mongod.conf"
+
+
+class T(component.T):
+    def internal_build_compile(self):
+        from rsconf import systemd
+
+        self.buildt.require_component("base_all")
+        self.j2_ctx, z = self.j2_ctx_init()
+        # TODO(e-carlin): this is a goofy catch 22
+        z.mongo_conf_path = self.j2_ctx.rsconf_db.host_run_d.join(self.name).join(
+            _MONGO_CONF_F
+        )
+        z.run_d = systemd.custom_unit_prepare(self, self.j2_ctx, (z.mongo_conf_path,))
+
+    def internal_build_write(self):
+        from rsconf import systemd
+        from rsconf.component import logrotate
+
+        z = self.j2_ctx.mongod
+        z.run_u = "mongod"
+        z.data_d = z.run_d.join(_DATA_DIR)
+        z.log_d = z.run_d.join(_LOG_DIR)
+        z.pid_d = pkio.py_path("/run/mongod")
+        z.pid_file_path = z.pid_d.join("mongod.pid")
```

Comment:
You call it `ip`, not `bind_ip`, but you call this `pid_file_path`. I realize you are trying to keep consistent names. I fee like we have a model here where we name things `_f` and should be consistent with that as opposed to be consistent with the files. Not sure, just an opinion.